client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.controller.js

```diff
@@ -0,0 +1,376 @@
+        return this.OvhApiCloudProjectSshKey.Lexi().save({ serviceName: this.serviceName }, this.newSshKey).$promise
+            .then(newSshKey => {
+                this.OvhApiCloudProjectSshKey.Lexi().resetQueryCache();
+                return this.$q.all({
+                    newSshKey,
+                    sshKeys: this.OvhApiCloudProjectSshKey.Lexi().query({ serviceName: this.serviceName }).$promise
+                });
+            })
+            .then(({ newSshKey, sshKeys }) => {
+                this.displayedSshKeys = sshKeys;
+                this.model.sshKey = newSshKey;
+                this.checkSshKeyByRegion(newSshKey.regions);
+            })
+            .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_sshKey_adding_ERROR"))
+            .finally(() => {
+                this.resetAddingSshKey();
+                this.loaders.addingSsh = false;
+            });
+    }
+
+    resetAddingSshKey () {
+        _.set(this.newSshKey, "name", null);
+        _.set(this.newSshKey, "publicKey", null);
+        this.addingSshKey = false;
+    }
+
+    /*-----------------------------------------
+     |  Step 2 : Region and DataCenter choice  |
+     -----------------------------------------*/
+
+    initRegionsAndDataCenters () {
+        _.set(this.loaders, "step2", true);
+        return this.$q.all({
+            regions: this.OvhApiCloudProjectRegion.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(regions => {
+                    this.regions = _.map(regions, region => this.RegionService.getRegion(region));
+                    return this.VirtualMachineAddService.getRegionsByImageType(this.regions, this.images, _.get(this.model, "imageType"));
+                }),
+            quota: this.promiseQuota
+                .then(quota => (this.quota = quota))
+                .catch(this.ServiceHelper.errorHandler("cpcivm_add_step2_quota_ERROR"))
+        })
+            .then(({ regions }) => {
+                _.forEach(regions, region => {
+                    // Add quota info
+                    this.RegionService.constructor.addOverQuotaInfos(region, this.quota);
+
+                    // Check SSH Key opportunity
+                    if (_.get(this.model, "sshKey.regions", false)) {
+                        this.RegionService.constructor.checkSshKey(region, this.model.sshKey.regions);
+                    }
+                });
+
+                this.displayedRegions = this.VirtualMachineAddService.groupRegionsByDatacenter(regions);
+                this.groupedRegions = _.groupBy(this.displayedRegions, "continent");
+            })
+            .catch(this.ServiceHelper.errorHandler("cpcivm_add_step2_regions_ERROR"))
+            .finally(() => {
+                this.loaders.step2 = false;
+            });
+    }
+
+    isStep2Valid () {
+        return this.model.region && this.model.imageId;
+    }
+
+    resetStep2 () {
+        _.set(this.model, "region", null);
```

Comment:
this.model.region = null.  this.model is always defined

Reply:
done